### PR TITLE
Add names to coroutines

### PR DESCRIPTION
**Pull request by shellderp**
Helps debugging in tests

### CODE REVIEW

File: service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchRunner.kt [hunk 1]
```diff
@@ -109,6 +110,7 @@ class BatchRunner(
       )
       rpcBackpressureChannel.upstream().send(Unit)
     }
+    rpcBackpressureChannel.upstream().close()
```
Comment (shellderp) [inline service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchRunner.kt:113]: technically this was missing but isn't really harmful either way..

Comment (mpawliszyn) [reply service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchRunner.kt:113]: Is this too aggressive? Will the awaiter be able to read from it after this still?

**Review by mpawliszyn — APPROVED** [review]
A small comment

File: service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchRunner.kt (hunk 1 above)
Comment (shellderp) [reply service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchRunner.kt:113]: `close()` doesn't prevent reading all the messages already sent. the receive() will just error on ClosedException once all messages are read

`close(exception)` will immediately cancel the channel and the receive() fails. before this change we were just cancelling all the coroutines from the BackfillRunner anyway, now it just closes first then cancels. I don't feel strongly but it's just technically correct to close all the channels you send on

**Pull request merged**